### DISCUSSION TITLE
Help with the the new Print_Stat Macro

### DISCUSSION THREAD

**Discussion by appleimperio** [jschuh/klipper-macros#116] [Q&A] [answered]
I'm trying to Dock the probe turn some lights and Homing the bed in the preheat phase and then I want it to continue probing and in the Print start phase-extruder  to dock the probe again. I'm sure I'm doing it wrong can you provide and example based on what I have. Because I need to make more changes. Thank You

````
M190 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
M109 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
_PRINT_START_PHASE_INIT EXTRUDER={first_layer_temperature[initial_tool]} BED=[first_layer_bed_temperature] MESH_MIN={first_layer_print_min[0]},{first_layer_print_min[1]} MESH_MAX={first_layer_print_max[0]},{first_layer_print_max[1]} LAYERS={total_layer_count} NOZZLE_SIZE={nozzle_diameter[0]}
; Insert custom gcode here.
_PRINT_START_PHASE_PREHEAT
NEVERMORE_ON
  G28 X 
  G90 
  G28 Y 
  G90 
  Attach_Probe_Lock
  G28 Z
  G90 
  PARK 
  STATUS_WLED_HEATING 
  STATUS_SB_HEATING ; LER Insert custom gcode here.
_PRINT_START_PHASE_PROBING
; Insert custom gcode here.
_PRINT_START_PHASE_EXTRUDER
Dock_Probe_Unlock ; LER Insert custom gcode here.
_PRINT_START_PHASE_PURGE

; This is the place to put slicer purge lines if you haven't set a non-zero
; variable_start_purge_length to have START_PRINT automatically calculate and 
; perform the purge (e.g. if using a Mosaic Palette, which requires the slicer
; to generate the purge).
````

**Comment by jschuh** [comment]
Could you attach your config? More specifically, I'd need to see the source to the `Attach_Probe_Lock` and `Dock_Probe_Unlock` macros, plus any additional overrides of `G28`, and I need to know if your `[stepper_z]` config includes a `probe:z_virtual_endstop`.

Or, assuming you're just pulling this config from another project, it's fine to just provide a link to their github/repo.

**Reply by appleimperio** [reply]
[printers_configs.zip](https://github.com/jschuh/klipper-macros/files/11566341/printers_configs.zip)
If you need anything else please let me know. Thank You

**Reply by jschuh** [reply, marked as answer]
I would think you want something like this, since you need to attach the probe before the first homing move (which occurs in the preheat phase) and then detach after the probing phase is complete:

```
M190 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
M109 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
_PRINT_START_PHASE_INIT EXTRUDER={first_layer_temperature[initial_tool]} BED=[first_layer_bed_temperature] MESH_MIN={first_layer_print_min[0]},{first_layer_print_min[1]} MESH_MAX={first_layer_print_max[0]},{first_layer_print_max[1]} LAYERS={total_layer_count} NOZZLE_SIZE={nozzle_diameter[0]}
STATUS_WLED_HEATING ; LER Insert custom gcode here.
STATUS_SB_HEATING ; LER Insert custom gcode here.
NEVERMORE_ON ; LER Insert custom gcode here.
G28 X Y ; LER Insert custom gcode here.
Attach_Probe_Lock ; LER Insert custom gcode here.
STATUS_WLED_HEATING ; LER Insert custom gcode here.
STATUS_SB_HEATING ; LER Insert custom gcode here.
_PRINT_START_PHASE_PREHEAT
; Insert custom gcode here.
_PRINT_START_PHASE_PROBING
Dock_Probe_Unlock ; LER Insert custom gcode here.
_PRINT_START_PHASE_EXTRUDER
; Insert custom gcode here.
_PRINT_START_PHASE_PURGE

; This is the place to put slicer purge lines if you haven't set a non-zero
; variable_start_purge_length to have START_PRINT automatically calculate and 
; perform the purge (e.g. if using a Mosaic Palette, which requires the slicer
; to generate the purge).
```
I don't think you needed any of those G90 commands; the klicky macros appear to set the necessary mode and use SAVE_GCODE_STATE/RESTORE_GCODE_STATE to restore it when done.

**Reply by jschuh** [reply]
I should mention that the approach above will cause the X and Y axes to home twice. If you want to avoid that you could just remove the `Attach_Probe_Lock` line from the slicer config above, and update line 566 in klippy-macros.cfg to this:

```
    {% if 'PROBE_LOCK' in params or printer.idle_timeout.state|string == "Printing" %}
```

Then the first homing move during a print will implicitly attach the probe, and it will stay attached until the later call to `Dock_Probe_Unlock` explicitly detaches it.

**Reply by appleimperio** [reply]
Perfect. Is there a way to make the extruder rise to the final temp after probing?

**Reply by jschuh** [reply]
That's what `_PRINT_START_PHASE_EXTRUDER` does.
